src/components/calculator.js

```diff
@@ -1,112 +1,108 @@
-import React from 'react';
+import React, { useState } from 'react';
 import calculate from '../logic/calculate';
 
-class Calculate extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      total: 0,
-      next: null,
-      operation: null,
-    };
-  }
+// refactor from class to function using hooks useState;
+const Calc = () => {
+  const [state, setState] = useState({
+    total: 0,
+    next: null,
+    operation: null,
+  });
 
-  handPress = (e) => {
-    const newState = calculate(this.state, e.target.innerText);
-    this.setState(newState);
+  const handPress = (e) => {
+    const newState = e.target.innerText;
+    setState((state) => calculate(state, newState));
   };
 
-  render() {
-    const { total, next, operation } = this.state;
+  const { total, next, operation } = state;
 
-    return (
-      <div className="calculator-box">
-        <table id="calculator" cellSpacing="0">
-          <tbody id="calculator-body">
-            <tr className="result-box">
-              <td colSpan="4" className="result">
-                <span>{total}</span>
-                <span>{next}</span>
-                <span>{operation}</span>
-              </td>
-            </tr>
+  return (
+    <div className="calculator-box">
+      <table id="calculator" cellSpacing="0">
+        <tbody id="calculator-body">
+          <tr className="result-box">
+            <td colSpan="4" className="result">
+              <span>{total}</span>
+              <span>{next}</span>
+              <span>{operation}</span>
```

Comment:
[OPTIONAL] I think it is a good idea to change the order of these elements to improve the user's experience :eyes: :eyes: . Currently, you are showing in the display something like + 9 8 To show in the display something like 8 + 9 the order should be the next:
<span>{next}</span>
<span>{operation}</span>
<span>{total}</span> 

![image](https://user-images.githubusercontent.com/98363075/202616572-d197db6a-e99b-414c-9e73-d9f21b3af55d.png)

![image](https://user-images.githubusercontent.com/98363075/202616661-8f6bc3a8-a479-4cff-866a-15dd9661597b.png)

Reply:
thank you so much changed it already